### PR TITLE
Trivial layout as staticmethod

### PR DESCRIPTION
**Pull request by 1ucian0**
In the effort to #1299, this PR introduces a static method for creating a default `Layout` and tries to unify the way that mappers initialize a layout.

### CODE REVIEW

File: qiskit/mapper/_layout.py [hunk 1]
```diff
@@ -69,6 +75,19 @@ def __delitem__(self, key):
         dict.__delitem__(self, self[key])
         dict.__delitem__(self, key)
 
+    @staticmethod
+    def checktype(thing):
```
Comment (ajavadia) [inline qiskit/mapper/_layout.py]: why does this need to be a staticmethod? a private method of self seems more appropriate.

File: qiskit/mapper/_layout.py [hunk 2]
```diff
@@ -69,6 +75,19 @@ def __delitem__(self, key):
...
+        """Checks if thing is a valid type"""
+        if thing is None:
+            return
+        if isinstance(thing, int):
+            return
+        if isinstance(thing, tuple) and \
+                len(thing) == 2 and \
+                isinstance(thing[0], Register) and isinstance(thing[1], int):
+            return
+        raise LayoutError('The element %s should be a Register or an integer' % (thing,))
```
Comment (ajavadia) [inline qiskit/mapper/_layout.py]: According to what the method is doing, this error message should be updated:
```suggestion
        raise LayoutError('The element %s should be a (Register, integer) tuple or an integer' % (thing,))
```

File: qiskit/mapper/_layout.py [hunk 3]
```diff
@@ -69,6 +75,19 @@ def __delitem__(self, key):
         dict.__delitem__(self, self[key])
         dict.__delitem__(self, key)
 
+    @staticmethod
+    def checktype(thing):
+        """Checks if thing is a valid type"""
+        if thing is None:
+            return
```
Comment (ajavadia) [inline qiskit/mapper/_layout.py]: why is None a valid input to `__setitem__`?

File: qiskit/mapper/_layout.py (hunk 1 above)
Comment (1ucian0) [reply qiskit/mapper/_layout.py]: Static methods also warrantee that they will not have side effects in `self`. `checktype` is pure with respect to self. I can rename it as `_checktype` if that preferable.

File: qiskit/mapper/_layout.py (hunk 3 above)
Comment (1ucian0) [reply qiskit/mapper/_layout.py]: Because it is convenient when Layouts is defined as `[(q0, 0), (q1, 0), None, (q1, 1), (q0, 1)]`.

**Pull request merged**